_rules/aria-role-permitted-j7zzqr.md

```diff
@@ -0,0 +1,252 @@
+	<li>
+		Apple
+		<hr role="presentation" />
+	</li>
+	<li>
+		Banana
+		<hr role="presentation" />
+	</li>
+	<li>Orange</li>
+</ul>
+```
+
+#### Passed Example 4
+
+This `p` element has an [explicit role][] of `generic` which is allowed. Using an explicit role of `generic` for any element is globally not recommended, but not an error by itself.
+
+```html
+<p role="generic">It was a dark and stormy night.</p>
+```
+
+#### Passed Example 5
+
+These `hr` elements have an [explicit role][] of `separator`, which is allowed. Using an explicit role which is the same as the [implicit role][] is not recommended, but not an error by itself.
+
+```html
+Fruits:
+<ul>
+	<li>
+		Apple
+		<hr role="separator" />
+	</li>
+	<li>
+		Banana
+		<hr role="separator" />
+	</li>
+	<li>Orange</li>
+</ul>
+```
+
+#### Passed Example 6
+
+This `h1` element has an [explicit role][] of `tab`, which is allowed. The `btn` token is not a valid role and is therefore ignored.
+
+```html
+<h1 role="btn tab">ACT rules</h1>
+```
+
+#### Passed Example 7
+
+This `dialog` element has an [explicit role][] of `alertdialog`, which is allowed. Even though the `alert` role is not allowed, the first token is a valid role and is therefore used.
+
+```html
+<dialog role="alertdialog alert">This is not right.</dialog>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `button` element has an [explicit role][] of `heading`, which is not allowed.
+
+```html
+<button role="heading">ACT rules</button>
+```
+
+#### Failed Example 2
+
+This `aside` element has an [explicit role][] of `navigation`, which is not allowed.
+
+```html
+<aside role="navigation">
+	<a href="https://www.w3.org">W3C</a> <a href="https://www.w3.org/TR/WCAG21/">WCAG 2.1</a>
+	<a href="https://www.w3.org/WAI/standards-guidelines/act/rules/"> ACT rules</a>
+</aside>
+```
+
+#### Failed Example 3
+
+These `h1` elements have an [explicit role][] of `listitem`, which is not allowed; the `div` element has an [explicit role][] of `list`, which is allowed.
+
+```html
+Fruits:
+<div role="list">
+	<h1 role="listitem">Apple</h1>
+	<h1 role="listitem">Banana</h1>
+	<h1 role="listitem">Orange</h1>
+</div>
+```
+
+#### Failed Example 4
+
+This `a` element with an `href` attribute has an [explicit role][] of `presentation`, which is not allowed. Since the element is focusable, the [presentational roles conflict resolution][] would trigger, and the explicit role is ignored.
+
+```html
+<a href="https://www.w3.org/WAI/standards-guidelines/act/rules/" role="presentation">All ACT rules</a>
+```
+
+#### Failed Example 5
+
+The first `li` element has an [explicit role][] of `presentation`, which is not allowed when its parent has an [implicit role][] of `list`.
+
+```html
+Fruits:
+<ul>
+	<li role="presentation">Apple</li>
+	<li>Banana</li>
+	<li>Orange</li>
+</ul>
+```
+
+#### Failed Example 6
+
+This `label` element has an [explicit role][] of `generic`, which is not allowed.
+
+```html
+<label role="generic"> Name: <input /> </label>
```

Comment:
ARIA is a little vague on whether generic elements are supposed to be in the accessibility tree. I think the applicability as you have it now probably means this passes. My suggested update to it should solve that.

Reply:
'passes' though a should not from aria.

i think aria in html's rules for label are a bit too strict here.  i'm going to make an issue to address this.